File: exercises/D701/solution.c
#include <stdio.h>
int q[1000];
char s[10];
int main() {
    int n, x, i;
    int qs = 0, qe = 0;
    scanf("%d", &n);
    for (i = 0; i < n; i++) {
        scanf("%s", s);
        if (strcmp(s, "PUSH") == 0) {
            scanf("%d", &x);
            q[qe++] = x;
        } else if (strcmp(s, "FRONT") == 0) {
            if (qs == qe) {
                printf("Empty\n");
            } else {
                printf("%d\n", q[qs]);
            }
        } else if (strcmp(s, "POP") == 0) {
            if (qs == qe) {
                printf("Cannot pop\n");
            } else {
                qs++;
            }
        } else {
            printf("%d\n", qe - qs);
        }
    }
    return 0;
}

File: exercises/D701/solution.pas
var
    q: array[1..1000] of longint;
    n, i, dummy, qs, qe: longint;
    s: string;
begin
    qs := 1;
    qe := 1;
    readln(n);
    for i := 1 to n do
    begin
        readln(s);
        if (copy(s, 1, 4) = 'PUSH') then
        begin
            val(copy(s, 6, 10), q[qe], dummy);
            inc(qe)
        end
        else if (s = 'FRONT') then
        begin
            if (qs = qe) then
                writeln('Empty')
            else
                writeln(q[qs])
        end
        else if (s = 'POP') then
        begin
            if (qs = qe) then
                writeln('Cannot pop')
            else
                inc(qs)
        end
        else
            writeln(qe - qs)
    end
end.
